beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/DepositProvider.java

```diff
@@ -122,7 +123,15 @@ public void onNewFinalizedCheckpoint(
       if (depositIndex.isGreaterThanOrEqualTo(finalizedState.getEth1Data().getDepositCount())
           && depositMerkleTree.getDepositCount()
               >= finalizedState.getEth1Data().getDepositCount().longValue()) {
-        depositMerkleTree.finalize(finalizedState.getEth1Data());
+        final Optional<UInt64> heightOptional =
+            eth1DataCache
+                .getEth1DataAndHeight(finalizedState.getEth1Data())
+                .map(Eth1DataCache.Eth1DataAndHeight::getBlockHeight);
+        if (heightOptional.isEmpty()) {
+          LOG.warn("Eth1Data height not found in cache. Skipping DepositTree finalization");
```

Comment:
Just a suggestion to tweak this wording so the user doesn't freak out and think Teku is unable to process finalisation of the chain.

```suggestion
          LOG.warn("Eth1Data height not found in cache. Skipping DepositTree pruning");
```

Reply:
That's fine, it just means that we don't prune the deposit tree that we store on disk or provide from the API (and if we've never been able to run the pruning we won't have a snapshot at all but that's a situation that has to be handled anyway).  Finalization is an important word to users about the chain advancing so they'll freak out with the current message.